ui/page/send/shared_widgets.go

```diff
@@ -0,0 +1,580 @@
+package send
+
+import (
+	"fmt"
+	"strings"
+
+	"gioui.org/layout"
+	"gioui.org/widget"
+
+	"github.com/crypto-power/cryptopower/app"
+	sharedW "github.com/crypto-power/cryptopower/libwallet/assets/wallet"
+	libUtil "github.com/crypto-power/cryptopower/libwallet/utils"
+	"github.com/crypto-power/cryptopower/ui/cryptomaterial"
+	"github.com/crypto-power/cryptopower/ui/load"
+	"github.com/crypto-power/cryptopower/ui/modal"
+	"github.com/crypto-power/cryptopower/ui/page/components"
+	"github.com/crypto-power/cryptopower/ui/utils"
+	"github.com/crypto-power/cryptopower/ui/values"
+)
+
+const (
+	// MaxTxLabelSize defines the maximum number of characters to be allowed on
+	// txLabelInputEditor component.
+	MaxTxLabelSize = 100
+)
+
+var (
+	automaticCoinSelection = values.String(values.StrAutomatic)
+	manualCoinSelection    = values.String(values.StrManual)
+)
+
+type sharedProperties struct {
+	*load.Load
+
+	parentWindow  app.WindowNavigator
+	pageContainer *widget.List
+
+	sourceWalletSelector  *components.WalletAndAccountSelector
+	sourceAccountSelector *components.WalletAndAccountSelector
+	sendDestination       *destination
+	amount                *sendAmount
+
+	infoButton    cryptomaterial.IconButton
+	retryExchange cryptomaterial.Button
+	nextButton    cryptomaterial.Button
+
+	shadowBox *cryptomaterial.Shadow
+	backdrop  *widget.Clickable
+
+	isFetchingExchangeRate bool
+	isModalLayout          bool
+
+	exchangeRate        float64
+	usdExchangeSet      bool
+	exchangeRateMessage string
+	confirmTxModal      *sendConfirmModal
+	currencyExchange    string
+
+	txLabelInputEditor cryptomaterial.Editor
+
+	*authoredTxData
+	selectedWallet  *load.WalletMapping
+	feeRateSelector *components.FeeRateSelector
+
+	toCoinSelection *cryptomaterial.Clickable
+
+	selectedUTXOs selectedUTXOsInfo
+}
+
+type authoredTxData struct {
+	destinationAddress  string
+	destinationAccount  *sharedW.Account
+	sourceAccount       *sharedW.Account
+	txFee               string
+	txFeeUSD            string
+	totalCost           string
+	totalCostUSD        string
+	balanceAfterSend    string
+	balanceAfterSendUSD string
+	sendAmount          string
+	sendAmountUSD       string
+}
+
+type selectedUTXOsInfo struct {
+	sourceAccount    *sharedW.Account
+	selectedUTXOs    []*sharedW.UnspentOutput
+	totalUTXOsAmount int64
+}
+
+func newSharedProperties(l *load.Load, isModalLayout bool) *sharedProperties {
```

Comment:
nit: I think this is just moved code, but lower case L is not a great variable name imo. It looks like 1 and I and l and | can make it `loader` or something?